src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3TestBase.cs

```diff
@@ -396,9 +397,11 @@ internal async Task SendDataAsync(Memory<byte> data, bool endStream = false)
             internal async Task<Dictionary<string, string>> ExpectHeadersAsync()
             {
                 var http3WithPayload = await ReceiveFrameAsync();
+                Assert.Equal(Http3FrameType.Headers, http3WithPayload.Type);
+
                 _decodedHeaders.Clear();
                 _qpackDecoder.Decode(http3WithPayload.PayloadSequence, this);
-                return _decodedHeaders;
+                return _decodedHeaders.ToDictionary(kvp => kvp.Key, kvp => kvp.Value, _decodedHeaders.Comparer);
```

Comment:
Why was this needed?

Reply:
Trailers will replace headers if a copy isn't made.

Reply:
Ah I see.